src/Core/functions.php

```diff
@@ -526,3 +531,88 @@ function toBool(mixed $value): ?bool
 
     return null;
 }
+
+/**
+ * Converts a value to a native DateTime object.
+ *
+ *  integer - value is treated as a Unix timestamp
+ *  string - value is treated as a ISO-8601 (Atom) formatted timestamp
+ *  Other values returns as null.
+ *
+ * @param mixed $value The value to convert to DateTime.
+ * @return \Cake\I18n\DateTime|null Returns a DateTime object if parsing is successful, or NULL otherwise.
+ * @since 5.1.0
+ */
+function toDateTime(mixed $value): ?DateTime
+{
+    if ($value instanceof DateTime) {
+        return $value;
+    }
+
+    if ($value instanceof DateTimeInterface) {
+        return DateTime::parse($value);
+    }
+
+    if (is_int($value)) {
+        try {
+            return DateTime::createFromTimestamp($value);
+        } catch (Exception) {
+            return null;
+        }
+    }
+
+    if (is_string($value)) {
+        try {
+            return DateTime::createFromFormat(DateTimeInterface::ATOM, $value);
+        } catch (Exception) {
+            return null;
+        }
+    }
+
+    return null;
+}
+
+/**
+ * Converts a value to a native Date object.
+ *
+ *  integer - value is treated as a Unix timestamp
+ *  string - value is treated as a I18N short formatted date
+ *  Other values returns as null.
+ *
+ * @param mixed $value The value to convert to Date.
+ * @return Date|null Returns a Date object if parsing is successful, or NULL otherwise.
+ * @since 5.1.0
+ */
+function toDate(mixed $value): ?Date
```

Comment:
Same here with having a format string with a default, so that userland code can do their thing as needed, and that we could use it in `Database\Type\DateType` as well.